### ISSUE TITLE
Recovery package has invalid composer.lock

### ISSUE THREAD

**Issue by evs-xsarus** [shopware/platform#2908]
### PHP Version

8.1

### Shopware Version

6.4.18

### Expected behaviour

A composer install without an error code as result.

### Actual behaviour

Composer returns error code 4 because the Recovery package has an invalid composer.lock .

NB Possible new behaviour since composer 2.5.x

A debug run shows:

> Reading ./composer.lock (/sites/sac/shopware/vendor/shopware/recovery/Common/composer.lock) Warning: The lock file is not up to date with the latest changes in composer.json....

And this is correct when I validate:

*/vendor/shopware/recovery/Common (master) $ composer validate*
```
./composer.json is valid for simple usage with Composer but has
strict errors that make it unable to be published as a package
See https://getcomposer.org/doc/04-schema.md for details on the schema
# Publish errors
- name : The property name is required
- description : The property description is required
# Lock file errors
- The lock file is not up to date with the latest changes in composer.json, it is recommended that you run `composer update` or `composer update <package name>`.
- Required package "doctrine/dbal" is in the lock file as "2.13.3" but that does not satisfy your constraint "~2.13.8".
This usually happens when composer files are incorrectly merged or the composer.json file is manually edited.
Read more about correctly resolving merge conflicts https://getcomposer.org/doc/articles/resolving-merge-conflicts.md
and prefer using the "require" command over editing the composer.json file directly https://getcomposer.org/doc/03-cli.md#require
# General warnings
- No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
- require.defuse/php-encryption : exact version constraints (2.2.1) should be avoided if the package follows semantic versioning
- require.hansott/psr7-cookies : exact version constraints (3.0.0) should be avoided if the package follows semantic versioning
- require.paragonie/random_compat : exact version constraints (2.0.18) should be avoided if the package follows semantic versioning
- require.pimple/pimple : exact version constraints (3.2.3) should be avoided if the package follows semantic versioning
- require.psr/log : exact version constraints (1.1.2) should be avoided if the package follows semantic versioning
- require.slim/php-view : exact version constraints (2.2.1) should be avoided if the package follows semantic versioning
- require.slim/slim : exact version constraints (3.12.3) should be avoided if the package follows semantic versioning
```

### How to reproduce

Use this CI/CD for gitlab:

```yaml
# Specify the correct PHP version used for your project.
image: thecodingmachine/php:8.1-v4-cli-node16

stages:
  - Build

composerShopware:
  stage: Build
  needs: []
  cache:
    key:
      files:
        - composer.json
        - composer.lock
      prefix: production
    paths:
      - vendor
    policy: pull-push
  tags:
    - linux
  script:
    - composer validate
    - COMPOSER_CACHE_DIR=composer-cache composer install --ignore-platform-reqs --optimize-autoloader --prefer-dist --no-dev --no-progress --no-suggest
    - find vendor -type d -name \.git -print0 | xargs -0 rm -rf
```

**Comment by evs-xsarus** [comment]
See a possible fix in this MR: https://github.com/shopware/recovery/pull/6

**Comment by shyim** [comment]
see https://github.com/shopware/production/issues/168#issuecomment-1359599136

its fixed already in trunk and 6.4 branch
